DevOps.Util/HelixServer.cs

```diff
@@ -62,38 +62,62 @@ public async ValueTask GetHelixPayloads(string jobId, List<string> workItems, st
 
-                    if (workItems[0] == "all")
-                    {
-                        downloadAll = true;
-                    }
+                bool downloadAll = false;
+                bool downloadFirst = workItems.Count == 0;
 
-                    foreach (WorkItemInfo workItemInfo in workItemsInfo)
-                    {
-                        if (!downloadAll && !workItems.Contains(workItemInfo.WorkItemId ?? string.Empty))
-                            continue;
+                if (!downloadFirst && workItems[0] == "all")
+                {
+                    downloadAll = true;
+                }
 
-                        if (string.IsNullOrEmpty(workItemInfo.PayloadUri))
-                            continue;
+                foreach (WorkItemInfo workItemInfo in workItemsInfo)
+                {
+                    if (!downloadFirst && !downloadAll && !workItems.Contains(workItemInfo.WorkItemId ?? string.Empty))
+                        continue;
 
-                        string fileName = Path.Combine(workItemsDir, $"{workItemInfo.WorkItemId}.zip");
+                    if (string.IsNullOrEmpty(workItemInfo.PayloadUri))
+                        continue;
 
-                        Console.WriteLine($"WorkItem {workItemInfo.WorkItemId} => {fileName}");
+                    await DownloadWorkitemFiles(workItemInfo.WorkItemId!, workItemInfo.PayloadUri).ConfigureAwait(false);
 
-                        await _client.DownloadZipFileAsync(workItemInfo.PayloadUri, fileName).ConfigureAwait(false);
-                    }
+                    // if no workitems specified, download the first one,
+                    // usefull to download any workitem and inspect the payload structure for debugging
+                    if (downloadFirst)
+                        return;
                 }
-                else
+
+                async Task DownloadWorkitemFiles(string workItemId, string payloadUri)
                 {
-                    throw new ArgumentException("No workitems specified");
+                    string itemDir = Path.Combine(workItemsDir, workItemId);
+                    Directory.CreateDirectory(itemDir);
+
+                    Console.WriteLine();
+                    Console.WriteLine($"------ Downloading files for: {workItemId} -------");
+                    Console.WriteLine();
+
+                    string fileName = Path.Combine(workItemsDir, itemDir, $"{workItemId}.zip");
+
+                    Console.WriteLine($"WorkItem {workItemId} => {fileName}");
+
+                    await _client.DownloadZipFileAsync(payloadUri, fileName, showProgress: true).ConfigureAwait(false);
+
+                    IEnumerable<UploadedFile> workitemFiles = await helixApi.WorkItem.ListFilesAsync(workItemId, jobId);
+                    foreach (var file in workitemFiles)
+                    {
+                        if (ignoreDumps && (file.Name.StartsWith("core.") || file.Name.EndsWith(".dmp")))
```

Comment:
Crazy nit pick: `StartsWith` and `EndsWith` should always have an explicit comparer